index-alphabetic-browse.sh

```diff
@@ -31,11 +31,24 @@ then
   SOLR_HOME="$VUFIND_HOME/solr/vufind"
 fi
 
+# This would be /opt/bitnami/solr on a solr docker container
+if [ -z "$SOLR_JAR_PATH" ]
+then
+  SOLR_JAR_PATH="${SOLR_HOME}/../vendor"
+fi
+
 set -e
 set -x
 
 cd "`dirname $0`/import"
-CLASSPATH="browse-indexing.jar:${VUFIND_HOME}/import/lib/*:${SOLR_HOME}/jars/*:${SOLR_HOME}/../vendor/modules/analysis-extras/lib/*:${SOLR_HOME}/../vendor/server/solr-webapp/webapp/WEB-INF/lib/*"
+LOG4J_CLASSPATH=$(find "${SOLR_JAR_PATH}/server/lib/ext" -name 'log4j*.jar' -printf '%p:' | sed 's/:$//')
```

Comment:
Why are we explicitly adding Log4j to the classpath? It doesn't seem that it was needed in the previous code. Is this dependency added as a result of introducing SolrMarc to the class path? If so, I'm a bit concerned, because SolrMarc currently uses Log4j 1.x and Solr uses Log4j 2.x. Also just wondering if there's an opportunity to simplify changes.

Reply:
Yes, it's a dependency. It crashes without Log4j on the classpath. I don't think we could simplify changes much. We could optimize further by checking which of the Log4j jars is really needed (I did not test that).
I did not realize the difference in versions, but since the classpath did not include Log4j before, it is safe to say that for this execution it will only be used by SolrMarc.